src/modules/household/components/householdActions/JoinHouseholdDialog.tsx

```diff
@@ -0,0 +1,277 @@
+import { MergeTypeRounded } from '@mui/icons-material';
+import { Typography } from '@mui/material';
+import React, { useCallback, useMemo, useState } from 'react';
+import Loading from '@/components/elements/Loading';
+import StepDialog, { TabDefinition } from '@/components/elements/StepDialog';
+import {
+  clientBriefName,
+  findHohOrRep,
+  sortHouseholdMembers,
+  stringifyHousehold,
+} from '@/modules/hmis/hmisUtil';
+import AddRelationshipsStep from '@/modules/household/components/householdActions/AddRelationshipsStep';
+import JoinHouseholdReview from '@/modules/household/components/householdActions/JoinHouseholdReview';
+import JoinHouseholdSelectClients from '@/modules/household/components/householdActions/JoinHouseholdSelectClients';
+import SuccessWayfindingStep from '@/modules/household/components/householdActions/SuccessWayfindingStep';
+import {
+  HouseholdClientFieldsFragment,
+  HouseholdFieldsFragment,
+  ProjectAllFieldsFragment,
+  RelationshipToHoH,
+  useGetEnrollmentWithHouseholdQuery,
+  useJoinHouseholdMutation,
+} from '@/types/gqlTypes';
+
+interface Props {
+  open: boolean;
+  conflictingEnrollmentId: string;
+  onClose: VoidFunction;
+  onComplete?: (
+    joinedHousehold: HouseholdFieldsFragment,
+    remainingHousehold?: HouseholdFieldsFragment | null
+  ) => void;
+  receivingHousehold: HouseholdFieldsFragment;
+  project: Pick<ProjectAllFieldsFragment, 'id' | 'projectName'>;
+}
+
+const JoinHouseholdDialog = ({
+  open,
+  onClose,
+  onComplete,
+  conflictingEnrollmentId,
+  receivingHousehold,
+  project,
+}: Props) => {
+  // Data this join workflow is collecting: What clients are joining and what their relationships are
+  const [joiningClients, setJoiningClients] = useState<
+    HouseholdClientFieldsFragment[]
+  >([]);
+  const [relationships, setRelationships] = useState<
+    Record<string, RelationshipToHoH | null>
+  >({});
+
+  // Fetch the conflicting enrollment by ID since we need the full EnrollmentWithHousehold
+  const {
+    data: { enrollment: conflictingEnrollment } = {},
+    loading: fetchLoading,
+    error: fetchError,
+  } = useGetEnrollmentWithHouseholdQuery({
+    variables: {
+      id: conflictingEnrollmentId,
+    },
+    onCompleted: (data) => {
+      if (data?.enrollment?.household) {
+        const household = data?.enrollment?.household;
+        const initiator = household?.householdClients.find(
+          (hc) => hc.enrollment.id === conflictingEnrollmentId
+        );
+
+        if (initiator) {
```

Comment:
What happens if `initiator` is undefined? maybe we should be raising instead if that happens?